internal/pkg/manifest/svc.go

```diff
@@ -582,16 +665,30 @@ func (rr *RoutingRule) Target(exposedPorts ExposedPortsIndex) (targetContainer s
 	targetContainer = exposedPorts.WorkloadName
 	targetPort = exposedPorts.mainContainerPort()
 
+	rrTargetContainer, rrTargetPort := rr.exposedContainerAndPort(exposedPorts)
+	if rrTargetContainer != nil {
+		targetContainer = aws.StringValue(rrTargetContainer)
+	}
+	if rrTargetPort != nil {
+		targetPort = aws.StringValue(rrTargetPort)
+	}
+	return
+}
+
+// exposedContainerAndPort returns the targetContainer and targetPort from a given ExposedPortsIndex for a routing rule.
+func (rr *RoutingRule) exposedContainerAndPort(exposedPorts ExposedPortsIndex) (*string, *string) {
+	var targetContainer, targetPort *string
+
 	rrTargetContainer := rr.TargetContainer
 	rrTargetPort := rr.TargetPort
 	if rrTargetContainer == nil && rrTargetPort == nil { // both targetPort and targetContainer are nil.
```

Comment:
Would ⬇️ work correctly? I am not super sure why we needed `rrTargetContainer` and `rrTargetPort` 🤔  I might have missed something.
```suggestion
	if rr.TargetContainer == nil && rr.TargetPort == nil { // both targetPort and targetContainer are nil.
```